### PR TITLE
[Mega-Feat]: Implement Group Endpoints

### LINKED ISSUE #314: [Mega-Feat] [Create CRUD, full endpoint for Group, add APIdocs, and also add testcases]
## 1. Create all endpoints for Group
Make a file in routes folder with the name `group.js`
make a URL as `group/create`, `group/list`, `group/update`, `group/delete`
then make the required files and functions in services and controller folder.
## 2. Create all testcases for Group
add testcases for all endpoints of group, inside `group.test.js` inside of `test/routes`
## 3. Create APIDOCs for Group
make api doc in `_apidoc.js` for all requests in the route for Group

### PR DESCRIPTION
**Pull request by sumitst05**
## Description

This PR adds the Group endpoints to the project. It includes the following changes:

- Created controllers and services for Group endpoint
- Added routes for Group operations
- Integrated Group routes in `app.js`
- Implemented test cases for all Group endpoints
- Documented the Group API using apidoc

## Fixes
#314 

## Checklist

- [x] Code follows project's style guidelines.
- [x] Changes are documented appropriately.

### CODE REVIEW

File: test/routes/group.test.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  cleanUp(done);
+});
+
+describe("group API", () => {
+  it("should create group", async () => {
+    const response = await agent.post("/group/add").send({
+      title: "Group 1",
+      student: "64fdc67feca8a69f01b33614",
+    });
+
```
Comment (Dhruvrg) [inline test/routes/group.test.js:47]: check content-type for post and update operations is it is json or not
eg :- expect(response.headers["content-type"]).toMatch(/json/);
refer accreditation.test.js

Comment (sumitst05) [reply test/routes/group.test.js]: Whoops! missed that...
Added the  necessary checks now :)
Thanks...

**Comment by Dhruvrg** [comment]
LGTM

**Review by TejasNair9977 — APPROVED** [review]
lgtm

**Pull request merged**